File: RedesII/practica1/servidor_tcp.py
# Servidor iterativo TCP en Python


from socket import *

serverName = '192.168.1.45'
serverPort = 12001
serverSocket = socket(AF_INET, SOCK_STREAM)
serverSocket.bind((serverName, serverPort))
serverSocket.listen(1)
print "Servidor preparado para recibir"
while 1:
	connectionSocket, addr = serverSocket.accept()
	sentence = connectionSocket.recv(1024)
	capitalizedSentence = sentence.upper()
	print "Voy a enviar el mensaje", capitalizedSentence
	connectionSocket.send(capitalizedSentence)
	connectionSocket.close()

File: RedesII/practica1/servidor_tcp.c
/* Servidor iterativo TCP */

int tcp_listen(char * arg1, char * arg2, socklen_t * addrlen);

int main(int argc, char **argv) {
	int listenfd, connfd;
	socklen_t clilen, addrlen;
	struct sockaddr *cliaddr;

	/* Contiene las llamadas a socket(), bind() y listen() */
	listenfd = tcp_listen(argv[1], argv[2], &addrlen);

	for ( ; ; ) {
		connfd = accept(listenfd, cliaddr, &clilen);
		process_request(connfd);

		Close(connfd);
	}
}

int tcp_listen(char * arg1, char * arg2, socklen_t * addrlen) {

}

